isolated-buttons/src/Button.js

```diff
@@ -0,0 +1,113 @@
+import {Observable, Subject} from 'rx'
+import isolate from '@cycle/isolate'
+import {div, button} from 'cycle-snabbdom'
+import combineLatestObj from 'rx-combine-latest-obj'
+
+const ADD = id => ({id, type: 'add'})
+const REMOVE = id => ({id, type: 'remove'})
+
+function intent({DOM, action$}, id) {
+  const add$ = DOM.select('.add').events('click').map(() => ADD(id))
+  const remove$ = DOM.select('.remove').events('click').map(() => REMOVE(id))
+  const removeButton$ = action$.filter(action => action.type === 'remove')
+  return {add$, remove$, removeButton$}
+}
+
+function findButton(children, _id) {
```

Comment:
I'm not that happy about binary search (also in other examples, it's my fault). I think specially for this example where new items are only inserted at the end (line 45), the array respects insertion order, so we could likely use a data structure like http://devdocs.io/javascript/global_objects/map ES6 Map which allows you to transform the entries into an array using `[...myMap]` but also allows you to directly find and remove an entry given its key. If immutability is a concern, then I'd use Immutable.js Map or OrderedMap.